lib/array_equals.rb

```diff
@@ -1,5 +1,24 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1 == nil && array2 == nil
+    return true
+  elsif array1 == nil || array2 == nil
+    return false
+  elsif array1.length != array2.length
+    return false
+  elsif array1.length == array2.length
+    element_count = 0
+    array1.length.times do |x|
+      if array1[x] == array2[x]
+        element_count += 1
```

Comment:
Keeping track of `element_count` is extra work. If the elements of the arrays (at the same index) are not equal, you should return false right away.